js/tests/test_examples.test.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * E2E Tests for plugin demos and examples.
+ */
+
+const { execFile } = require('child_process');
+const path = require('path');
+const { promisify } = require('util');
+const execFileAsync = promisify(execFile);
+
+// Paths
+const PLUGINS = path.resolve(__dirname, '../plugins');
+const EXAMPLES_PATH = path.resolve(__dirname, '../examples');
+
+// Require env vars
+const { COMPOSIO_API_KEY, OPENAI_API_KEY } = process.env;
+
+// Plugin test definitions
+const EXAMPLES = {
+  portfolio_generator: {
+    file: path.join(EXAMPLES_PATH, 'portfolio-generator', 'demo.mjs'),
+    match: {
+      type: 'stdout',
+      values: ['🎉Output from agent: '],
+    },
+    env: { COMPOSIO_API_KEY },
+  },
```

Comment:
Tested this? This IMO should fail if other model key is not provided.

Reply:
We are passing it from the `process.env`, This is working